### PR TITLE
Improve `hostInfo` command's `os` response

### LINKED ISSUE #447: Improve `hostInfo` command's `os` response
We could improve our handling of `hostInfo` command by adding `os.name` and `os.version` fields as best effort. That's ok to do that only on Linux.

The simplest way to do that seems to be to parse `/etc/os-release` file. More info: https://www.freedesktop.org/software/systemd/man/os-release.html

### PR DESCRIPTION
**Pull request by DoodgeMatvey**
Closes #447.

### CODE REVIEW

**Review by seeforschauer — CHANGES_REQUESTED** [review]
@DoodgeMatvey tests are failing, please update `go mod`

File: internal/handlers/pg/msg_hostinfo.go [hunk 1]
```diff
@@ -22,6 +22,8 @@ import (
 	"strings"
 	"time"
 
+	"gopkg.in/ini.v1"
```
Comment (AlekSi) [inline internal/handlers/pg/msg_hostinfo.go:25]: Let's not use an additionaly dependency for something simple as this task

Comment (DoodgeMatvey) [reply internal/handlers/pg/msg_hostinfo.go]: Fixed

**Review by seeforschauer — CHANGES_REQUESTED** [review]
can you please add some tests? 
also, the parsing host info func itself can be moved to common functions, to be reusable by other handlers which may appear in the future.

**Comment by AlekSi** [comment]
I'm all for adding tests with `/etc/os-release` file content from different Linux distribution, but I'm not sure what common function could be extracted there. We could always extract common code when it is, well, common.

**Review by AlekSi — CHANGES_REQUESTED** [review]
See the comment above

File: integration/parse_test.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+ID_LIKE=debian
+PRETTY_NAME=Ubuntu 18.04 LTS
+VERSION_ID=18.04
+HOME_URL=https://www.ubuntu.com/
+SUPPORT_URL=https://help.ubuntu.com/
+BUG_REPORT_URL=https://bugs.launchpad.net/ubuntu/
+PRIVACY_POLICY_URL=https://www.ubuntu.com/legal/terms-and-policies/privacy-policy
+VERSION_CODENAME=bionic
+UBUNTU_CODENAME=bionic
+`,
```
Comment (AlekSi) [inline integration/parse_test.go:55]: Can we get one more from Red Hat / RPM-based distro?

File: internal/handlers/common/parse.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import (
+	"bufio"
+	"io"
+	"strings"
+
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// ParseOSRelease parses the etc/os-release file.
+func ParseOSRelease(reader io.Reader) (map[string]string, error) {
```
Comment (AlekSi) [inline internal/handlers/common/parse.go:26]: I think that function could return just name and version, so caller should not have to know what valid keys are

Comment (AlekSi) [reply internal/handlers/common/parse.go:26]: Also, let's move it to msg_hostinfo.go and unexport

File: integration/parse_test.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+			"PRIVACY_POLICY_URL": "https://www.ubuntu.com/legal/terms-and-policies/privacy-policy",
+			"VERSION_CODENAME":   "bionic",
+			"UBUNTU_CODENAME":    "bionic",
+		},
+	}
+
+	for key, testCase := range testCases {
+		osReleaseReader := bytes.NewReader([]byte(osReleaseFilesData[key]))
+
+		osRelease, err := common.ParseOSRelease(osReleaseReader)
```
Comment (AlekSi) [inline integration/parse_test.go:92]: That's not an integartion test, that's [unit test](https://github.com/FerretDB/FerretDB/blob/main/CONTRIBUTING.md#code-overview), it should be in `internal/`

We should also update this test to check that name and version are set on Linux: https://github.com/FerretDB/FerretDB/blob/3b082cd08832682aee502f7b972742cbc35c71fb/integration/commands_diagnostic_test.go#L44

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: internal/handlers/common/parse.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	scanner := bufio.NewScanner(reader)
+
+	configParams := map[string]string{}
+
+	for scanner.Scan() {
+		str := strings.Split(scanner.Text(), "=")
+
+		configParams[str[0]] = str[1]
```
Comment (seeforschauer) [inline internal/handlers/common/parse.go:34]: too many line breaks here and it seems it's made not to logically separate, WDYT of changing this?

Comment (DoodgeMatvey) [reply internal/handlers/common/parse.go:34]: I didn't catch what you mean. Do you mean indent or what?

File: integration/parse_test.go (hunk 2 above)
Comment (DoodgeMatvey) [reply integration/parse_test.go]: Yes, added

File: internal/handlers/common/parse.go (hunk 3 above)
Comment (DoodgeMatvey) [reply internal/handlers/common/parse.go]: Now `ParseOSRelease` return just name and version and moved to `msg_hostinfo.go` but in order to be able to use this function in tests, I left it exported

File: integration/parse_test.go (hunk 4 above)
Comment (DoodgeMatvey) [reply integration/parse_test.go]: Fixed

File: internal/handlers/parse_osrelease_test.go [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+		},
+	}
+
+	for key, testCase := range testCases {
+		osReleaseReader := bytes.NewReader([]byte(osReleaseFilesData[key]))
+
+		osName, osVersion, err := pg.ParseOSRelease(osReleaseReader)
+		require.NoError(t, err)
+
+		assert.Equal(t, testCase["NAME"], osName, "expected: %s\nactual  : %s", testCase["NAME"], osName)
```
Comment (AlekSi) [inline internal/handlers/parse_osrelease_test.go:98]:
```suggestion
		assert.Equal(t, testCase["NAME"], osName)
```

The standard message should be good enogh, no?

File: internal/handlers/pg/msg_hostinfo.go [hunk 7]
```diff
@@ -60,3 +77,26 @@ func (h *Handler) MsgHostInfo(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg
 
 	return &reply, nil
 }
+
+// ParseOSRelease parses the etc/os-release file.
+func ParseOSRelease(reader io.Reader) (string, string, error) {
```
Comment (AlekSi) [inline internal/handlers/pg/msg_hostinfo.go:82]: We should not have that function and unit test for it in different packaages.

I think the best place for both could be internal/handlers/common

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: internal/handlers/pg/msg_hostinfo.go (hunk 7 above)
Comment (DoodgeMatvey) [reply internal/handlers/pg/msg_hostinfo.go:82]: If I place them in `internal/handlers/common` and make it unexportable I cannot use it in pg package for `MsgHostInfo` func

File: internal/handlers/parse_osrelease_test.go (hunk 6 above)
Comment (DoodgeMatvey) [reply internal/handlers/parse_osrelease_test.go]: Elsewhere in the code I have seen `"expected: %s\nactual  : %s"`, but what you suggested is also ok, I replaced

File: internal/handlers/pg/msg_hostinfo.go (hunk 7 above)
Comment (DoodgeMatvey) [reply internal/handlers/pg/msg_hostinfo.go]: I placed them in `internal/handlers/common` but left the function exported. What do you think about it?

File: internal/handlers/common/parse.go (hunk 5 above)
Comment (seeforschauer) [reply internal/handlers/common/parse.go]: I meant empty lines, a lines w/o code

**Review by seeforschauer — CHANGES_REQUESTED** [review]
lgfm, just some formatting

File: internal/handlers/common/parse_osrelease.go [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// ParseOSRelease parses the etc/os-release file.
+func ParseOSRelease(reader io.Reader) (string, string, error) {
+	scanner := bufio.NewScanner(reader)
+
+	configParams := map[string]string{}
+
```
[the remark this answers is not in the document]
Comment (DoodgeMatvey) [reply internal/handlers/common/parse_osrelease.go]: fixed

File: internal/handlers/common/parse_osrelease.go [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+	for scanner.Scan() {
+		str := strings.Split(scanner.Text(), "=")
+
```
[the remark this answers is not in the document]
Comment (DoodgeMatvey) [reply internal/handlers/common/parse_osrelease.go]: fixed

File: internal/handlers/common/parse_osrelease.go [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+		if len(str) == 1 {
+			continue
+		}
+
```
[the remark this answers is not in the document]
Comment (DoodgeMatvey) [reply internal/handlers/common/parse_osrelease.go]: fixed

File: internal/handlers/common/parse_osrelease.go [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+		configParams[str[0]] = str[1]
+	}
+
```
[the remark this answers is not in the document]
Comment (DoodgeMatvey) [reply internal/handlers/common/parse_osrelease.go]: Thank you, fixed

**Review by seeforschauer — APPROVED** [review]
Thank you!

**Review by AlekSi — APPROVED** [review]

**Pull request merged**